routers/api/v1/misc/nodeinfo.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package misc
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
+)
+
+// NodeInfo returns the NodeInfo for the Gitea instance to allow for federation
+func NodeInfo(ctx *context.APIContext) {
+	// swagger:operation GET /nodeinfo miscellaneous getNodeInfo
+	// ---
+	// summary: Returns the nodeinfo of the Gitea application
+	// produces:
+	// - application/json
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/NodeInfo"
+
+	nodeInfo := &structs.NodeInfo{
+		Version: "2.1",
+		Software: structs.NodeInfoSoftware{
+			Name:       "gitea",
+			Version:    setting.AppVer,
+			Repository: "https://github.com/go-gitea/gitea.git",
```

Comment:
That will be outdated as soon as `Gitea` will be migrated to `gitea.com`

Reply:
I think we can change it after migration, now I think it's okay.